### PR TITLE
FOUR-15984: FOUR-15515: A flag needs to saved when the abe was executed

### PR DESCRIPTION
**Pull request by luNunezProcessmaker**
## Issue & Reproduction Steps
When a request was executed by action by email the following column needs to populated with 
 `process_request_tokens.is_actionbyemail = 1`
It means that the Request was closed by action by email actions

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-15984

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

ci:deploy
ci:next
ci:screen-builder:feature/FOUR-15510
ci:connector-send-email:feature/FOUR-15510
ci:processmaker-bpmn-moddle:feature/FOUR-15510

### CODE REVIEW

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-fb8fca10d5.engk8s.processmaker.net

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 1]
```diff
@@ -780,4 +780,20 @@ public function screenRequested(Request $httpRequest, ProcessRequest $request)
...
+     * @return Response
+     */
+    public function abeFlag($id)
+    {
+        $query = ProcessRequestToken::query();
+        $query->where('id', $id)
+            ->where('status', 'ACTIVE')
+            ->update(['is_actionbyemail' => true]);
+            
+        return response()->json([], 200);
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:797]: Return a boolean please

File: ProcessMaker/Http/Controllers/Api/ProcessRequestController.php [hunk 2]
```diff
@@ -780,4 +780,20 @@ public function screenRequested(Request $httpRequest, ProcessRequest $request)
...
         return new ApiCollection($response);
     }
+
+    /**
+     * Adding abe flag
+     * @param  int  $id
+     * 
+     * @return Response
+     */
+    public function abeFlag($id)
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/Api/ProcessRequestController.php:790]: rename this function to enableIsActionbyemail

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Repositories/TokenRepository.php [hunk 3]
```diff
@@ -161,6 +157,16 @@ public function persistActivityActivated(ActivityInterface $activity, TokenInter
         $token->getInstance()->updateCatchEvents();
         $token->saveOrFail();
         $token->setId($token->getKey());
+
+        // Check if is script or self service again to send the Action by email with the updated token
+        if (!$isScriptOrServiceTask) {
+            $this->validateAndSendActionByEmail($activity, $token, $user->email);
+            $isActionsByEmail = $activity->getProperty('isActionsByEmail', false);
+            if ($isActionsByEmail) {
+                (new ProcessRequestController)->abeFlag($token->getKey());
```
Comment (pmPaulis) [inline ProcessMaker/Repositories/TokenRepository.php:166]: please move the update is_Actionbyemail `TaskController::updateVariable`

**Review by pmPaulis — CHANGES_REQUESTED** [review]

File: ProcessMaker/Http/Controllers/TaskController.php [hunk 4]
```diff
@@ -226,7 +227,10 @@ public function updateVariable(HttpRequest $request, $abe_uuid)
                         $task,
                         $data
                     );
-                    // Set here the flag is_actionbyemail
+
+                    // Set the flag is_actionbyemail
+                    (new ProcessRequestController)->abeFlag($task->id);
```
Comment (pmPaulis) [inline ProcessMaker/Http/Controllers/TaskController.php:232]: the function is given an error enableIsActionbyemail

**Review by pmPaulis — CHANGES_REQUESTED** [review]

**Comment by tokensPM** [comment]
QA server K8S was successfully deployed https://ci-fb8fca10d5.engk8s.processmaker.net

**Pull request merged**